src/App.tsx

```diff
@@ -15,49 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type Goods = string;
+type Reverse = boolean;
+
+enum Sortfield {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = '',
+}
+
+type Props = {
+  goods: Goods[];
+};
+
+const GoodsList: React.FC<Props> = ({ goods }) => (
+  <ul>
+    {goods.map(good => (
+      <li data-cy="Good" key={good}>
+        {good}
+      </li>
+    ))}
+  </ul>
+);
+
+function getPrepareGoods(
+  goods: Goods[],
+  sortGood: Sortfield | undefined,
+  reverse: Reverse,
+): Goods[] {
+  const prepareGoods: Goods[] = [...goods];
+
+  if (prepareGoods) {
```

Comment:
this condition will always return true because you already applied `prepareGoods` as [...goods]
arrays is always be true: [] === true or [{}] === true
in this case, you can check a prepareGoods.length